lib/dal/src/schema.rs

```diff
@@ -415,4 +450,27 @@ impl Schema {
         }
         Ok(None)
     }
+
+    /// Collect all [`FuncIds`](crate::Func) corresponding to the provided [`SchemaId`](Schema).
+    /// Since [`SchemaVariants`](SchemaVariant) can use the same [`Funcs`](crate::Func) (and
+    /// often do), we use a [`HashSet`] to de-duplicate results.
+    pub async fn all_func_ids(ctx: &DalContext, id: SchemaId) -> SchemaResult<HashSet<FuncId>> {
+        let mut func_ids = HashSet::new();
+        for schema_variant_id in Self::list_schema_variant_ids(ctx, id).await? {
+            func_ids.extend(
+                SchemaVariant::all_func_ids(ctx, schema_variant_id)
+                    .await
+                    .map_err(Box::new)?,
+            );
+        }
+        Ok(func_ids)
+    }
+
+    /// Collect all [`Funcs`](crate::Func) corresponding to the provided [`SchemaId`](Schema).
+    pub async fn all_funcs(ctx: &DalContext, id: SchemaId) -> SchemaResult<Vec<Func>> {
+        let func_id_set = Self::all_func_ids(ctx, id).await?;
+        let func_ids = Vec::from_iter(func_id_set.into_iter());
+        let funcs = Func::list_from_ids(ctx, func_ids.as_slice()).await?;
+        Ok(funcs)
+    }
```

Comment:
This allows us to get all funcs for all versions of an asset. This allows us to show all funcs for an asset in the "ASSET FUNCTIONS" section of the frontend.